app/src/main/java/org/greenstand/android/TreeTracker/messages/announcementmessage/AnnouncementScreen.kt

```diff
@@ -0,0 +1,249 @@
+            )
+        },
+        bottomBar = {
+            ActionBar(
+                leftAction = {
+                    ArrowButton(
+                        isLeft = true,
+                        colors = AppButtonColors.MessagePurple,
+                        onClick = {
+                            navController.popBackStack()
+                        }
+                    )
+                }
+            )
+        }
+    ) {
+        Column(
+            Modifier
+                .padding(top = 4.dp, start = 4.dp, end = 4.dp, bottom = 80.dp)
+                .fillMaxSize()
+                .nestedScroll(scrollBehavior.nestedScrollConnection)
+        ) {
+            Messages(
+                state = state,
+                modifier = Modifier.weight(1f),
+                scrollState = scrollState,
+                viewModel = viewModel
+            )
+            if (state.showNoInternetDialog) {
+                NoInternetDialog(viewModel = viewModel)
+            }
+        }
+    }
+}
+
+@Composable
+fun Messages(
+    state: AnnouncementState,
+    scrollState: LazyListState,
+    modifier: Modifier = Modifier,
+    viewModel: AnnouncementViewModel
+) {
+    Box(modifier = modifier) {
+        LazyColumn(
+            reverseLayout = true,
+            state = scrollState,
+            contentPadding = PaddingValues(top = 10.dp),
+            // Add content padding so that the content can be scrolled (y-axis)
+            // below the status bar + app bar
+            // TODO: Get height from somewhere
+            modifier = Modifier
+                .fillMaxSize()
+        ) {
+            items(state.messages) { message ->
+                ChatItemBubble(message = message, state = state, viewModel = viewModel)
+                Spacer(modifier = Modifier.height(6.dp))
+            }
+        }
+    }
+}
+
+@Composable
+fun ChatItemBubble(
+    message: AnnouncementMessage,
+    state: AnnouncementState,
+    viewModel: AnnouncementViewModel
+) {
+    val context = LocalContext.current
+
+    Column(
+        modifier = Modifier
+            .padding(start = 10.dp, end = 60.dp)
+            .background(
+                color = AppColors.MessageReceivedBackground,
+                shape = RoundedCornerShape(6.dp)
+            )
+            .fillMaxWidth()
+            .wrapContentHeight()
+            .padding(8.dp), horizontalAlignment = Alignment.Start
+    ) {
+        message.body?.let {
+            Text(
+                text = it,
+                color = CustomTheme.textColors.lightText,
+                fontWeight = FontWeight.Bold,
+                style = CustomTheme.typography.regular,
+            )
+        }
+        Spacer(modifier = Modifier.height(4.dp))
+        message.videoLink?.let {
+            Text(
+                text = it,
+                Modifier
+                    .padding(top = 10.dp)
+                    .clickable(onClick = {
+                        if (state.isInternetAvailable) {
```

Comment:
@ZavenArra Should we check for internet before clicking a link? I think we should always open it. If there is no internet, the browser will let the user know.

Reply:
i think it would be better to check for internet, since our users are not that familiar with mobile devices.  as first time users, they could get somewhat lost if we take them out of the app for no reason.  let's make it as easy as possible for offline use and not offer a click on a link if they are offline, since this will be a lot of time time.

Reply:
However when I tested this, I was online and still got the internet not available message when clicking on the link.

Reply:
@ZavenArra what test email handle did you use?(that contains a url). I haven't been able to test it myself